test/testdata/lsp/fast_path/crash_local_vars.1.rbupdate

```diff
@@ -0,0 +1,11 @@
+# typed: strict
+
+class Outer
+  describe 'client' do
+    #               ^^ error: Hint: this kDO_BLOCK token might not be properly closed
+    class Inner <
+    it 'foo' do # error: Superclasses must only contain constant literals
+      1.times {|x| puts(x)}
+    end
+  end
+end # error: unexpected token "end of file"
```

Comment:
> Jake, why is this a fast_path test? It looks like a crash that has nothing to do with LSP.

So the thing is: the crash only shows up in `Substitute.cc`. That only fires when there are multiple GlobalState's to merge, but we don't have that in PosTests because we intentionally limit the execution to be single threaded.

There is a second way to trigger `Substitute.cc`, even if the test is single threaded, which is by making a test that calls `commitEdit`, because then Sorbet's LSP will try to see if the edit can take the fast path or not by running index + substutition + hashing.

This is also the code path the bug manifested in the wild, but that's just because most people don't tend to try to commit code where a class has a method + block call as an ancestor in a file that parses.